beta/src/components/MDX/ExpandableExample.tsx

```diff
@@ -17,15 +19,28 @@ interface ExpandableExampleProps {
 }
 
 function ExpandableExample({children, excerpt, type}: ExpandableExampleProps) {
-  const [isExpanded, setIsExpanded] = React.useState(false);
-  const isDeepDive = type === 'DeepDive';
-  const isExample = type === 'Example';
-
   if (!Array.isArray(children) || children[0].type.mdxName !== 'h4') {
     throw Error(
       `Expandable content ${type} is missing a corresponding title at the beginning`
     );
   }
+  const isDeepDive = type === 'DeepDive';
+  const isExample = type === 'Example';
+  const id = children[0].props.id;
+
+  const {asPath} = useRouter();
+  // use mounted state here incase of browser goBack
+  const [isMounted, setMounted] = useState(false);
+  // init as expanded to prevent flash
+  const [isExpanded, setIsExpanded] = useState(true);
+
+  // asPath would mismatch between server and client, reset here instead of put it into init state
+  useEffect(() => {
+    if (!isMounted) {
+      setMounted(true);
+      setIsExpanded(id === asPath.split('#')[1]);
```

Comment:
We should only do these state updates if we the condition actually matched. Otherwise, don't do anything. This prevents unnecessary work.